src/htm/ntypes/ArrayBase.cpp

```diff
@@ -525,31 +525,6 @@ std::istream &operator>>(std::istream &inStream, ArrayBase &a) {
   return inStream;
 }
 
-// a helper function to parse the type for an SDR in a JSON serialization of an Array.
-//  syntax "SDR(dim1[, dim2[, dim3]])"
-static std::vector<UInt> parseDim(const std::string &type) {
-  std::vector<UInt> dim;
-  char *end;
-  const char *p1 = strchr(type.c_str(), '(');
-  if (p1) {
-    const char *p2 = strchr(p1, ')');
-    if (p2) {
-      while (true) {
-        while (p1 < p2 && !isdigit(*p1))
-          p1++;
-        if (p1 >= p2)
-          break;
-        UInt i = strtoul(p1, &end, 0);
-        dim.push_back(i);
-        p1 = end;
-      }
-    }
-  }
-  NTA_CHECK(dim.size() > 0)
-      << "In parse of Array object from JSON, SDR type does not include dimensions.  Expected SDR(nnn[,nnn[,nnn]])";
-  return dim;
-}
-
 // Serialization and Deserialization using YAML parser
 // For JSON, expecting something like {type: "Int32", data: [1, 0, 1]}
 void ArrayBase::fromYAML(const std::string &data) { // handles both YAML and JSON
```

Comment:
This function is a low level function that is used for more than REST Api.  Let's leave it as it was, including the parseDim( ) function, and create a new function with the following signature.
```
    void ArrayBase::fromVaue(const Value& vm)
```
Then in RESTapi::put_input_request( ... )
- confirm that the input has been initialized...or just call itr->second.net->initialize( )
- fetch the Input's pre-allocated buffer  (the Array object)
- parse the input string into a Value object.
- verify that the number of data elements is consistent with the input's dimensions.  ( getCount( ) ).  Throw an exception if they don't match.
- call this new fromValue( ) function on the buffer to directly set its value.

The user does not need to pass in the type or the dimensions.